pkg/redfish/rfcomponents.go

```diff
@@ -1327,6 +1346,29 @@ func (s *EpSystem) discoverRemotePhase1() {
 						errlog.Printf("ERROR: unexpected type/value '%T'/'%v' detected for PowerConsumedWatts, setting to 0\n", pwrCtl.PowerConsumedWatts, pwrCtl.PowerConsumedWatts)
 					}
 				}
+				if IsManufacturer(s.SystemRF.Manufacturer, FoxconnMfr) == 1 {
+					// Even though we've successfully read the /Power endpoint, we have
+					// observed that the Paradise BMC fw may not have populated it with
+					// all of the data that we depend on if the node was just powered on.
+					// There is sometimes a lag after we receive the node power on event,
+					// and when /Power is correctly populated.  We check for what we
+					// depend upon here, and retry the /Power read after a short delay if
+					// we find any data missing.  The delay/retry count were selected based
+					// upon emperical observation.  If all the retries fail, just log an
+					// error and continue.
+					if powerRetryCount == 4 && (pwrCtl.PowerConsumedWatts == nil || pwrCtl.PowerCapacityWatts == 0) {
+						errlog.Printf("Foxconn Paradise WARNING: /Power endpoint not ready (%v, %d), retry %d in %d seconds\n", pwrCtl.PowerConsumedWatts, pwrCtl.PowerCapacityWatts, FoxconnPowerRetryNum, FoxconnPowerRetryDelay)
+						time.Sleep(time.Duration(FoxconnPowerRetryDelay) * time.Second)
+						if FoxconnPowerRetryNum == powerRetryCount {
```

Comment:
With all the gotos and risks to future code changes this should be changed to
if FoxconnPowerRetryNum >= powerRetryCount

Reply:
Per prior comment, fixed this by using isFoxconnPowerOnEventDiscovery boolean.